cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java

```diff
@@ -233,17 +234,20 @@ private void ensureTopicsCreated(Map<String, ?> config) {
    * Increase the replication factor of a Kafka topic through adding new replicas in a rack-aware, round-robin way.
    * If Zookeeper does not have rack information about brokers, then rack-aware property is not guaranteed.
    * In this case it is only guaranteed that new replicas are added to brokers which do not currently host the partition.
+   *
    * @param zkUtils ZkUtils class to use to increase replication factor.
    * @param topicMetadata Topic metadata stored in Zookeeper.
    * @param replicationFactor The replication factor to set for the topic.
    * @param topic The topic to apply the change.
-   * @param props The properties to set for the topic.
    */
-  private void increaseTopicReplicationFactor(ZkUtils zkUtils,
-                                              MetadataResponse.TopicMetadata topicMetadata,
-                                              int replicationFactor,
-                                              String topic,
-                                              Properties props) {
+  private void maybeIncreaseTopicReplicationFactor(ZkUtils zkUtils,
+                                                   MetadataResponse.TopicMetadata topicMetadata,
+                                                   int replicationFactor,
+                                                   String topic) {
+    if (!ensureNoPartitionUnderPartitionReassignment(zkUtils)) {
+      LOG.info("There are ongoing partition reassignments, skip checking topic {}'s replication factor.", topic);
```

Comment:
Nit: `topic {}'s replication factor.` -> `replication factor of topic {}.` -- i.e. I feel like it is better to avoid `'` in log messages.